hash_practice/exercises.py

```diff
@@ -2,18 +2,57 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
+        Time Complexity: O(n)
+        Space Complexity: O(n)
+"""
+    hash_table = {}
+    for word in strings:
+        # sort each char of word into a list
+        sorted_word = sorted(word)
+        # join the sorted chars back into a word
+        anagram = ''.join(sorted_word)
+        if anagram in hash_table:
+            # if anagram is already a key in hash table,
+            # append the current word to the value
+            hash_table[anagram].append(word)  
+        else:
+            # else, assign anagram as a key in the hash table
+            # and the current word as its value in list form
+            hash_table[anagram] = [word]
+    print(hash_table)     
+    # return just the list of all the values
+    return list(hash_table.values())
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 However because you're sorting, I think the time complexity is O(n log n) worst-case